tests/vcs/git/GitCommitBearTest.py

```diff
@@ -267,6 +269,149 @@ def test_body_checks(self):
+
+        # No keywords and no issues...
+        self.git_commit('Shortlog\n\n'
+                        'This line is ok.\n'
+                        'This line is by far too long (in this case).\n'
+                        'This one too, blablablablablablablablabla.')
+        self.assertEqual(self.run_uut(body_close_issue=True,),
+                         ['No issue reference found in the body at '
+                          'HEAD commit.'])
+        self.assert_no_msgs()
+
+        # Has keyword but no issue number...
+        self.git_commit('Shortlog\n\n'
+                        'First line, blablablablablabla.\n'
+                        'Another line, blablablablablabla.\n'
+                        'Fix #01112')
+        self.assertEqual(self.run_uut(
+                            body_close_issue=True,),
+                         ['No short issue reference found in the body '
+                          'at HEAD commit.'])
+
+        # GitHub host with short issue tag
+        self.git_commit('Shortlog\n\n'
+                        'First line, blablablablablabla.\n'
+                        'Another line, blablablablablabla.\n'
+                        'Fix #1112')
+        self.assertEqual(self.run_uut(
+                            body_close_issue=True,), [])
+
+        # GitHub host with no issue reference
+        self.git_commit('Shortlog\n\n'
+                        'First line, blablablablablabla.\n'
+                        'Another line, blablablablablabla.\n'
+                        'Fix #01112')
+        self.assertEqual(self.run_uut(
+                            body_close_issue=True,
+                            body_close_issue_full_url=True),
+                         ['No issue reference found in the '
+                          'body at HEAD commit.'])
+
+        # Adding GitLab remote for testing
+        self.run_git_command('remote', 'set-url', 'test',
+                             'git@gitlab.com:user/repo.git')
+
+        # GitLab chosen and has an issue
+        self.git_commit('Shortlog\n\n'
+                        'First line, blablablablablabla.\n'
+                        'Another line, blablablablablabla.\n'
+                        'Closing https://gitlab.com/user/repo/issues/1112')
+        self.assertEqual(self.run_uut(
+                            body_close_issue=True,
+                            body_close_issue_full_url=True), [])
+
+        # Invalid issue number in URL
+        self.git_commit('Shortlog\n\n'
+                        'First line, blablablablablabla.\n'
+                        'Another line, blablablablablabla.\n'
+                        'Closing https://gitlab.com/user/repo/issues/1af121')
+        self.assertEqual(self.run_uut(
+                            body_close_issue=True,
+                            body_close_issue_full_url='full_ref'),
+                         ['Invalid issue number present in the body '
+                          'at HEAD commit: 1af121'])
```

Comment:
yes please use 7 chars, this is a valid short-form git sha and does not confuse :)